src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -76,11 +77,23 @@
  * depending on the characteristics of the byte buffer instance the segment is associated with. For instance, a buffer memory
  * segment obtained from a byte buffer created with the {@link ByteBuffer#allocateDirect(int)} method will be backed
  * by native memory.
+ *
+ * <h2>Mapping memory segments from files</h2>
+ *
+ * It is also possible to obtain a native memory segment backed by a memory-mapped file using the factory method
+ * {@link MemorySegment#mapFile(Path, long, long, FileChannel.MapMode, ResourceScope)}. Such native memory segments are
+ * called <em>mapped memory segments</em>; mapped memory segments are associated with an underlying file descriptor.
  * <p>
- * Finally, it is also possible to obtain a memory segment backed by a memory-mapped file using the factory method
- * {@link MemorySegment#mapFile(Path, long, long, FileChannel.MapMode, ResourceScope)}. Such memory segments are called <em>mapped memory segments</em>;
- * mapped memory segments are associated with an underlying file descriptor. For more operations on mapped memory segments, please refer to the
- * {@link MappedMemorySegments} class.
+ * Contents of mapped memory segments can be {@link #force() persisted} and {@link #load() loaded} to and from the underlying file;
+ * these capabilities are suitable replacements for some of the functionality in the {@link java.nio.MappedByteBuffer} class.
+ * Note that, while it is possible to map a segment into a byte buffer (see {@link MemorySegment#asByteBuffer()}),
+ * and then call e.g. {@link java.nio.MappedByteBuffer#force()} that way, this can only be done when the source segment
+ * is small enough, due to the size limitation inherent to the ByteBuffer API.
+ * <p>
+ * Clients requiring sophisticated, low-level control over mapped memory segments, should consider writing
+ * custom mapped memory segment factories; using JNI, e.g. on Linux, it is possible to call {@code mmap}
```

Comment:
Can we replace the reference to JNI with the linker API?

Reply:
Sure - that's actually what I wanted to do in the first place - dunno how I ended up with JNI :-)